test/api/v3/integration/groups/POST-groups_groupId_leave.js

```diff
@@ -122,11 +122,15 @@ describe('POST /groups/:groupId/leave', () => {
         privateGuild = group;
         leader = groupLeader;
         invitedUser = invitees[0];
+
+        await leader.post(`/groups/${group._id}/chat`, { message: 'Some message' });
       });
 
-      it('removes a group when the last member leaves', async () => {
+      it('removes a group and any new messages when the last member leaves', async () => {
         await leader.post(`/groups/${privateGuild._id}/leave`);
 
+        expect(leader.newMessages).to.be.empty;
```

Comment:
You don't get notified for your own messages, so this isn't testing anything. Also, the leader object needs to sync.